frontend/src/api/prometheus/usePrometheusQueryRange.ts

```diff
@@ -22,10 +30,18 @@ const usePrometheusQueryRange = (
         query: `query=${queryLang}&start=${start}&end=${endInS}&step=${step}`,
       })
 
-      .then((response) => response.data?.response.data.result?.[0]?.values || []);
-  }, [queryLang, apiPath, span, endInMs, step]);
+      .then((response) => {
+        let result: T[] | PrometheusQueryRangeResultValue[];
+        if (responsePredicate) {
+          result = responsePredicate(response.data?.response.data);
+        } else {
+          result = response.data?.response.data.result?.[0]?.values || [];
+        }
+        return result as T[];
```

Comment:
@andrewballantyne I've fixed this by making the ResponsePredicate mandatory instead of optional, and creating a ResponsePredicate for the other prometheus hook. If you could take a look and tell me WDYT I'd appreciate it!

Relevant updates:
* https://github.com/alexcreasy/odh-dashboard/blob/31b47f84a4d0ad9864513212e38fe21b60700711/frontend/src/api/prometheus/usePrometheusQueryRange.ts#L33
* https://github.com/alexcreasy/odh-dashboard/blob/31b47f84a4d0ad9864513212e38fe21b60700711/frontend/src/api/prometheus/useQueryRangeResourceData.ts#L19